_articles/datadriven-ability-breakdown-documentation.md

```diff
@@ -222,6 +222,205 @@ Added in Reborn:
+* DOTA_UNIT_TARGET_FLAG_NOT_MAGIC_IMMUNE_ALLIES
+* DOTA_UNIT_TARGET_FLAG_NOT_ATTACK_IMMUNE
+* DOTA_UNIT_TARGET_FLAG_FOW_VISIBLE
+* DOTA_UNIT_TARGET_FLAG_INVULNERABLE
+* DOTA_UNIT_TARGET_FLAG_NOT_ANCIENTS
+* DOTA_UNIT_TARGET_FLAG_NOT_CREEP_HERO
+* DOTA_UNIT_TARGET_FLAG_NOT_DOMINATED
+* DOTA_UNIT_TARGET_FLAG_NOT_ILLUSIONS
+* DOTA_UNIT_TARGET_FLAG_NOT_NIGHTMARED
+* DOTA_UNIT_TARGET_FLAG_NOT_SUMMONED
+* DOTA_UNIT_TARGET_FLAG_OUT_OF_WORLD
+* DOTA_UNIT_TARGET_FLAG_PLAYER_CONTROLLED
+* DOTA_UNIT_TARGET_FLAG_PREFER_ENEMIES
+
+### Fun with Flags
+
+Flags were seen as AbilityUnitTargetFlags completions, but this is not their sole application.
+
+The same applies to Team and Types.
+
+ * `"Flags"` and `"ExcludeFlags"` in a `"Target"` block gives control over how to target units to apply actions on them later:
+ 
+~~~
+"Target"
+{
+    "Center"    "CASTER"
+    "Flags"     "DOTA_UNIT_TARGET_FLAG_DEAD"
+}
+~~~
+
+ * `"TargetFlags"` in a `"LinearProjectile"` action allows a `LinearProjectile` to ignore units that would otherwise be included by default in the Team+Type values, for example those with `MODIFIER_STATE_INVISIBLE`.
+ * `"Aura_Flags"` in a modifier with the other `"Aura"` keys can be used, for example, to make an [aura modifier](http://web.archive.org/web/20181130135800/http://moddota.com/forums/discussion/comment/29#Comment_29) only affect ranged units by adding `DOTA_UNIT_TARGET_FLAG_RANGED_ONLY`.
+ 
+The same applies for **Teams** and **Types**.
+
+*Example*: Targets all friendly units in a radius of the caster, including couriers, buildings, and siege units. Excludes heroes, summons, and other player controlled units.
+
+~~~
+"Target"
+{
+    "Center"        "CASTER"
+    "Radius"        "%radius"
+    
+    // AbilityUnitTargetTeam values.
+    "Teams"         "DOTA_UNIT_TARGET_TEAM_FRIENDLY"
+    
+    // AbilityUnitTargetTypes
+    "Types"         "DOTA_UNIT_TARGET_ALL"
+    "ExcludeTypes"  "DOTA_UNIT_TARGET_HERO"
+    
+    // AbilityUnitTargetFlags
+    "Flags"         "DOTA_UNIT_TARGET_FLAG_NOT_SUMMONED"
+    "ExcludeFlags"  "DOTA_UNIT_TARGET_FLAG_PLAYER_CONTROLLED"
+}
+~~~
+
+*Example*: Mirana's Arrow projectile rewrite that only hits heroes, including those that are magic immune:
+
+~~~
+"LinearProjectile"
+{
+    "Target"            "POINT"
+    "EffectName"        "particles/units/heroes/hero_mirana/mirana_spell_arrow.vpcf"
+    "MoveSpeed"         "857"
+    "StartRadius"       "115"
+    "EndRadius"         "115"
+    "StartPosition"     "attach_attack1"
+    "FixedDistance"     "3000"
+    "TargetTeams"       "DOTA_UNIT_TARGET_TEAM_ENEMY"
+    "TargetTypes"       "DOTA_UNIT_TARGET_HERO"
+    "TargetFlags"       "DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES"
+    "HasFrontalCone"    "0"
+    "ProvidesVision"    "1"
+    "VisionRadius"      "650"
+}
+~~~
+
+With `DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES`, and with `DOTA_UNIT_TARGET_FLAG_NONE`: [comparison gfy](https://gfycat.com/floweryunevenhorseshoebat).
```

Comment:
You can embed gfy content with `{% include gfycat.html id="floweryunevenhorseshoebat" %}`